packages/itwinui-react/src/core/Popover/Popover.tsx

```diff
@@ -120,16 +120,19 @@ type PopoverInternalProps = {
    * When additional parameters are _required_ for an interaction/trigger, an object must be passed to enable it.
    * Booleans will not be allowed in this case.
    *
+   * When trying to add `listNavigation`, consider using `useListNavigationProps` to generate the necessary props.
```

Comment:
Any reason we can't call this hook automatically like the other interactions?

Reply:
I don't think adding the `useEffect` and `getFocusableNodes` to `usePopover` is acceptable, since it's a hack that needs to be removed in the future (as discussed [previously](https://github.com/iTwin/iTwinUI/pull/2021#discussion_r1591006224)). Keep the hacks confined to `Menu` and/or a separate hook. I was just asking a question here, not making a suggestion.

Reply:
Ohh, I see. Reverted those changes to keep that `useEffect`/`getFocusableNodes` in `Menu` (https://github.com/iTwin/iTwinUI/pull/2021/commits/cffb1d0595c96bda2af2a7f8476e56f6c65c89c2).

> Any reason we can't handle this automatically based on the object passed to `listNavigation`?

In `usePopover`, I tried wrapping the `listNavigationProps` passed by `Menu` with useListNavigationProps()`. However, I could not get the arrow key navigation working correctly when doing this. I guess this has something to do with the stale `listRef` values being used since `listRef` a ref.

Thus, in that same commit, I reverted it such that `useListNavigationProps` needs to be called by `Menu` and its result passed to `usePopover`; and `listRef` needs to be updated by `Menu`.

Reply:
> However, I could not get the arrow key navigation working correctly when doing this. I guess this has something to do with the stale `listRef` values being used since `listRef` a ref.

Did you try using state instead of ref? Then it should be up-to-date, as it will re-render when the state changes.

Reply:
Yes, I tried it for a while initially but went the ref route since FloatingUI was expecting a ref for `listRef` ([docs](https://floating-ui.com/docs/useListNavigation#props)). I tried to make `useListNavigationProps` expose a custom `setListRef` function, but that caused some TS type complexities since `useListNavigationProps` was supposed to return `UseListNavigationProps`.

I will take another look at using states soon.

Reply:
A ref object is nothing special; it might be enough to just do something like this to satisfy the floating-ui API:

```js
useListNavigation(context, {
  listRef: { current: listElements },
});
```

where `listElements` is a prop stored in state.

Reply:
Two small suggestions:
1. Inside the hook, we should probably use `useLatestRef` to grab the latest instance of the filter function. Otherwise a new a mutation observer will be created on every single render.
2. (optional) The filter function can be made more intuitive to use. Try to avoid positional arguments to make the code more readable. The code below shows an object as the second argument of the hook and it lifts the `.filter` call to avoid the need for two arguments in the filter function.
   ```tsx
   const focusableElementsRef = useFocusableElementsRef(
     menuElement,
     {
       // Filter out focusable elements that are inside each menu item, e.g. checkbox, anchor
       filter: (allElements) => allElements.filter(
         (i) => !allFocusableElements?.some((p) => p.contains(i.parentElement)),
       ),
     },
   );
   ```

Also, should we move `useFocusableElementsRef` into `utils` into `focusable.ts` since we already have a file for it?

Reply:
the import in `focusable.ts` needs to be fixed to use namespace import for react (`import * as React`) and to import `useSyncExternalStore` from `utils` instead of react